packages/design-system/src/components/OcAvatars/OcAvatars.vue

```diff
@@ -157,6 +164,18 @@ const getAvatarComponentForItem = (item: Item) => {
 const avatarsClasses = computed(() => {
   return [`oc-avatars-gap-${getSizeClass(gapSize)}`, ...(stacked ? ['oc-avatars-stacked'] : [])]
 })
+
+onMounted(() => {
+  if (!unref(avatarsRef)) {
+    return
+  }
+
+  const avatarElements = Array.from(unref(avatarsRef).children)
+
+  avatarElements.forEach((child, index) => {
```

Comment:
We can avoid the repaint. You can give the desired z-index as property to all the avatars individually and then inside oc-avatar use the prop for a css variable.

Reply:
I am not sure if this is possible, since we have a slot for #userAvatars 🤔